rasa/core/training/interactive.py

```diff
@@ -919,43 +908,35 @@ def _intents_from_messages(messages):
 
 async def _correct_wrong_nlu(
     corrected_nlu: Dict[Text, Any],
-    evts: List[Dict[Text, Any]],
+    events: List[Dict[Text, Any]],
     endpoint: EndpointConfig,
     sender_id: Text,
 ) -> None:
     """A wrong NLU prediction got corrected, update core's tracker."""
 
-    latest_message = latest_user_message(evts)
-    corrected_events = all_events_before_latest_user_msg(evts)
-
-    latest_message["parse_data"] = corrected_nlu
-
-    await replace_events(endpoint, sender_id, corrected_events)
-
-    await send_message(
+    revert_latest_user_utterance = UserUtteranceReverted().as_dict()
+    listen_for_next_message = ActionExecuted(ACTION_LISTEN_NAME).as_dict()
+    corrected_message = latest_user_message(events)
+    corrected_message["parse_data"] = corrected_nlu
+    await send_event(
         endpoint,
         sender_id,
-        latest_message.get("text"),
-        latest_message.get("parse_data"),
+        [revert_latest_user_utterance, listen_for_next_message, corrected_message],
```

Comment:
Just for my understanding: The bot is listening and the user types something. We interpreted the user's message as intent A. However, that is wrong. Thus, we revert everything until the action listen, replay the action listen and replay the corrected user message. Correct?

Reply:
yes. `until action listen` includes the `action_listen`, hence we have to replay it.